cheat/src/utils/hook_system/mod.rs

```diff
@@ -69,59 +71,29 @@ impl Hook {
     /// # Panics
     ///
     /// Panics if it fails to lock the `TARGETS` mutex.
-    #[must_use]
-    pub fn hook(target: *const c_void, detour: *const c_void) -> bool {
+    pub fn hook(target: *const c_void, detour: *const c_void) -> anyhow::Result<()> {
         let Ok(mut targets) = TARGETS.lock() else {
             tracing::error!("failed to lock TARGETS");
-            return false;
+            bail!("failed to lock TARGETS");
         };
 
-        let mut hk =
-            Self { target: target.cast_mut(), detour: detour.cast_mut(), original: null_mut() };
+        let hook = unsafe { retour::RawDetour::new(target as _, detour as _)? };
```

Comment:
This can also be wrapped in [ManuallyDrop](https://doc.rust-lang.org/std/mem/struct.ManuallyDrop.html) instead of storing it in the struct below